### PR TITLE
Service factory

### PR DESCRIPTION
**Pull request by CatHuyuH24**
- Create a Service Factory implementing Singleton and Factory design pattern to centralize db-connection-instaces
- Update the source code to use the service factory instead of calling the constructor directly for Brand, Employee, Category, Product, Stripe, Settings, ConfigHelper, URILauncher, GoogleOAuth, EncryptionService
-  Perform tests to ensure old functions still work as expected
image

### CODE REVIEW

File: POSSystem/Models/Employee.cs
Comment (eNKay2408) [inline POSSystem/Models/Employee.cs:1]: Have you consider the Google OAuth login? In this case, the password is null. So this part can throw null exception

File: POSSystem/Services/ServiceFactory.cs
Comment (eNKay2408) [inline POSSystem/Services/ServiceFactory.cs:1]: - Although I cannot understand completely, it look very logical and clean.
- Thanks for the refactoring and enhancing maintenance

**Review by eNKay2408 — APPROVED** [review]
- So great. We need to improve scalability by implementing more design patterns such as you did. 
- Singleton and Factory are very important now to optimize the program

File: POSSystem/Models/Employee.cs
Comment (CatHuyuH24) [reply POSSystem/Models/Employee.cs:1]: My bad! Will fix ASAP!

Comment (CatHuyuH24) [reply POSSystem/Models/Employee.cs:1]: **Fixed in the update-ux-handling branch**

Comment (eNKay2408) [reply POSSystem/Models/Employee.cs:1]: Okay, nice bro

**Pull request merged**